.github/copilot-instructions.md

```diff
@@ -0,0 +1,193 @@
+
+```yaml
+pool:
+  name: "my-agent-pool"
+  demands:
+    - agent.name -equals blue-agent
+```
+
+### KEDA Auto-scaling
+
+- Include KEDA ScaledJob configuration in Kubernetes examples
+- Use Azure Pipelines scaler for auto-scaling
+- Set appropriate polling intervals (default: 10 seconds)
+- Example KEDA trigger:
+
+```yaml
+triggers:
+  - type: azure-pipelines
+    metadata:
+      organizationURL: "https://dev.azure.com/myorg"
+      personalAccessToken: "pat_token"
+      poolName: "my-pool"
+```
+
+### Pipeline Security
+
+- Use service connections instead of hardcoded credentials
+- Enable pipeline signing and verification
+- Use Azure AD authentication where possible
+- Implement least-privilege access for pipeline agents
+
+## Deployment and Tooling Preferences
+
+### Helm Preference
+
+- Prefer Helm over kubectl for all Kubernetes deployments
+- Use `helm install` and `helm upgrade` commands in documentation
+- Never suggest `kubectl apply` for application deployments
+- Example: `helm install blue-agent clemlesne-blue-agent/blue-agent --values values.yaml`
+
+### Bicep for Azure Resources
+
+- Use Bicep templates for Azure resource deployments
+- Prefer Bicep over ARM templates or Terraform
+- Include proper parameter validation and descriptions
+- Example: `az deployment sub create --template-file main.bicep --parameters @parameters.json`
+
+### Configuration Management
+
+- Use Helm values files for configuration management
+- Support both values.yaml and values.json formats
+- Implement proper parameter validation in Helm charts
+- Use ConfigMaps for non-sensitive configuration
+
+## Container and Image Standards
+
+### Base Image Selection
+
+- Prefer official base images from Microsoft, Red Hat, or Canonical
+- Support multiple OS flavors: azurelinux3, bookworm, jammy, noble, ubi8, ubi9
+- Use minimal/slim variants when available
+- Include Windows support with ltsc2019 and ltsc2022 variants
+
+### Container Labels
+
+- Include OCI-compliant labels in all container images
+- Add ArtifactHub metadata labels for discoverability
+- Use semantic versioning for image tags
+- Example labels:
+
+```dockerfile
+LABEL org.opencontainers.image.title="Blue Agent"
+LABEL org.opencontainers.image.description="Self-hosted Azure Pipelines agent"
+LABEL org.opencontainers.image.vendor="clemlesne"
+```
+
+### Container Registry
+
+- Support both GitHub Container Registry (ghcr.io) and Docker Hub
+- Use multi-registry publishing for redundancy
+- Implement proper image tagging strategy (latest, version, SHA)
+- Include registry authentication in CI/CD examples
+
+## Documentation Standards
+
+### Hugo Documentation
+
+- Use Hugo for all documentation generation
+- Follow existing documentation structure and themes
+- Include front matter with proper metadata
+- Use shortcodes for consistent formatting
+- Example front matter:
+
+```yaml
+---
+title: "Page Title"
+weight: 1
+prev: /previous-page
+next: /next-page
+---
+```
```

Comment:
Specific. That this documentation is then compiled into a website and accessible from Files, search engines and hyper easily develop to find documentation, examples of skates, configuration, but also advice on performance, configuration, deployment and use.